sdk/monitor/Azure.Monitor.OpenTelemetry.AspNetCore/tests/Azure.Monitor.OpenTelemetry.AspNetCore.Tests/AzureSdkLoggingTests.cs

```diff
@@ -86,18 +86,62 @@ public async Task PublicLogForwarderIsAdded(LogLevel eventLevel, string expected
             }
         }
 
-        private void WaitForRequest(MockTransport transport)
+        [Fact]
+        public async Task SelfDiagnosticsIsDisabled()
         {
+            var enableLevel = LogLevel.Debug;
+
+            var builder = WebApplication.CreateBuilder();
+            var transport = new MockTransport(_ => new MockResponse(200).SetContent("ok"));
+            builder.Logging.ClearProviders();
+            bool logAzureFilterCalled = false;
+            builder.Logging.AddFilter((name, level) =>
+            {
+                if (name != null && name.StartsWith("Azure"))
+                {
+                    logAzureFilterCalled = true;
+                    return level >= enableLevel;
+                }
+                return false;
+            });
+            builder.Services.AddOpenTelemetry().UseAzureMonitor(config =>
+            {
+                config.Transport = transport;
+                config.ConnectionString = $"InstrumentationKey={Guid.NewGuid()}";
```

Comment:
Lets please use;
```suggestion
                config.ConnectionString = "InstrumentationKey=00000000-0000-0000-0000-000000000000";
```

Although very rare, we don't want to collide with an actual user's Ikey.

Reply:
your transports are cached and static. They need random ikey to be distinguishable between tests ;)

Reply:
Ah, i didn't know that. In that case, I propose a prefix to the guid:

```suggestion
                config.ConnectionString = $"InstrumentationKey=UNITTEST_{Guid.NewGuid()}";
```